### PR TITLE
[docs] Revamp README and Ray intro doc page

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This PR revamps and aligns the README and Ray intro doc page:
- New "What is Ray" diagram that introduces AIR vs Ray core (diagram TBD finalized, this is the working placeholder)
- Update the description of Ray
- Link out to the user guides for key libraries and key concepts
- Remove old / broken links, as well as the inline library descriptions from the README

air1

### CODE REVIEW

**Comment by pcmoritz** [comment]
Small nit, I think it would be wonderful if the Ray library boxes were at the same height and the same size as the "Custom Libraries and Applications" box :)

File: README.rst [hunk 1]
```diff
@@ -14,381 +14,62 @@
...
 - `RLlib paper`_
-- `RLlib flow paper`_
 - `Tune paper`_
 
 *Older documents:*
 
 - `Ray paper`_
 - `Ray HotOS paper`_
 
+.. _`Ray AIR`: https://docs.ray.io/en/master/ray-air/getting-started.html
```
Comment (pcmoritz) [inline README.rst:62]: These currently all link to `master`. We might want to change to `latest` after 2.0 gets released to avoid confusion.

**Review by pcmoritz — APPROVED** [review]
You might consider bringing back the integrations in some way (e.g. having a link to the integrations from the main documentation landing page). After the revamp they are lost but are a very important part of the story.

File: README.rst [hunk 2]
```diff
@@ -14,381 +14,62 @@
...
+.. image:: https://github.com/ray-project/ray/raw/master/doc/source/images/what-is-ray-padded.svg
 
-Ray is packaged with the following libraries for accelerating machine learning workloads:
+..
+  https://docs.google.com/drawings/d/1Pl8aCYOsZCo61cmp57c7Sja6HhIygGCvSZLi_AuBuqo/edit
 
+Learn more about `Ray AIR`_ and its libraries:
+
+- `Datasets`_: Distributed Data Loading and Compute
+- `Train`_: Distributed Deep Learning
```
Comment (richardliaw) [inline README.rst:27]: this should be Distributed Training (since it now supports xgboost, lightgbm)

File: README.rst [hunk 3]
```diff
@@ -14,381 +14,62 @@
...
-**Ray provides a simple, universal API for building distributed applications.**
+.. image:: https://github.com/ray-project/ray/raw/master/doc/source/images/what-is-ray-padded.svg
 
-Ray is packaged with the following libraries for accelerating machine learning workloads:
+..
+  https://docs.google.com/drawings/d/1Pl8aCYOsZCo61cmp57c7Sja6HhIygGCvSZLi_AuBuqo/edit
 
+Learn more about `Ray AIR`_ and its libraries:
+
+- `Datasets`_: Distributed Data Loading and Compute
```
Comment (stephanie-wang) [inline README.rst:26]:
```suggestion
- `Datasets`_: Distributed Data Loading and Preprocessing
```

"Compute" sounds pretty vague to me.

File: README.rst [hunk 4]
```diff
@@ -14,381 +14,62 @@
...
-
-As well as libraries for taking ML and distributed apps to production:
-
 - `Serve`_: Scalable and Programmable Serving
-- `Workflow`_: Fast, Durable Application Flows (alpha)
 
-There are also many `community integrations <https://docs.ray.io/en/master/ray-libraries.html>`_ with Ray, including `Dask`_, `MARS`_, `Modin`_, `Horovod`_, `Hugging Face`_, `Scikit-learn`_, and others. Check out the `full list of Ray distributed libraries here <https://docs.ray.io/en/master/ray-libraries.html>`_.
+Or more about `Ray Core`_ and its key abstractions:
+
+- `Tasks`_: Stateless tasks executed in the cluster.
```
Comment (stephanie-wang) [inline README.rst:34]:
```suggestion
- `Tasks`_: Stateless functions executed in the cluster.
```

File: README.rst [hunk 5]
```diff
@@ -14,381 +14,62 @@
...
+- `Actors`_: Stateful worker processes created in the cluster.
+- `Objects`_: Shared-memory objects accessible across the cluster.
```
Comment (stephanie-wang) [inline README.rst:36]:
```suggestion
- `Objects`_: Shared-memory immutable values accessible across the cluster.
```
(to differentiate from Python objects)

File: README.rst [hunk 6]
```diff
@@ -14,381 +14,62 @@
...
 More Information
 ----------------
 
 - `Documentation`_
-- `Tutorial`_
-- `Blog`_
-- `Ray 1.0 Architecture whitepaper`_ **(new)**
-- `Exoshuffle: large-scale data shuffle in Ray`_ **(new)**
+- `Ray Architecture whitepaper`_
+- `Exoshuffle: large-scale data shuffle in Ray`_
```
Comment (stephanie-wang) [inline README.rst:53]: Add a link to the ownership paper?

File: doc/source/index.md [hunk 7]
```diff
@@ -83,17 +83,25 @@ or [Slurm](cluster/slurm) clusters.
...
+- [Datasets](data/dataset): Distributed Data Loading and Compute
+- [Train](train/train): Distributed Deep Learning
+- [Tune](tune/index): Scalable Hyperparameter Tuning
+- [Serve](serve/index): Scalable and Programmable Serving
+- [RLlib](rllib/index): Scalable Reinforcement Learning
+
+Or more about [Ray Core](ray-core/walkthrough) and its key abstractions:
+- [Tasks](ray-core/tasks): Stateless tasks executed in the cluster.
+- [Actors](ray-core/actors): Stateful worker processes created in the cluster.
+- [Objects](ray-core/objects): Shared-memory objects accessible across the cluster.
```
Comment (stephanie-wang) [inline doc/source/index.md:104]: Same changes suggested here.

File: README.rst (hunk 1 above)
Comment (ericl) [reply README.rst:62]: I changed them now (they all work).

**Review by stephanie-wang — APPROVED** [review]
Should also change the tagline on the main datasets docs page from "Compute" -> "Preprocessing".

**Comment by ericl** [comment]
> Should also change the tagline on the main datasets docs page from "Compute" -> "Preprocessing".

Yup I'll open a separate PR for this.

**Review by dmatrix — COMMENTED** [review]
Accept for the nit on the diagram. LGTM

**Review by dmatrix — APPROVED** [review]

**Pull request merged**